airflow/contrib/kubernetes/pod_launcher.py

```diff
@@ -83,16 +83,34 @@ def run_pod(self, pod, startup_timeout=120, get_logs=True):
     def _monitor_pod(self, pod, get_logs):
         # type: (Pod) -> (State, content)
 
+        status = None
         if get_logs:
-            logs = self._client.read_namespaced_pod_log(
-                name=pod.name,
-                namespace=pod.namespace,
-                container='base',
-                follow=True,
-                tail_lines=10,
-                _preload_content=False)
-            for line in logs:
-                self.log.info(line)
+            try:
+                logs = self._client.read_namespaced_pod_log(
+                    name=pod.name,
+                    namespace=pod.namespace,
+                    container='base',
+                    follow=True,
+                    tail_lines=10,
+                    _preload_content=False,
+                    _request_timeout=30
+                )
+
+                for line in logs:
+                    self.log.info(line)
+
+                    # try to get status ASAP. hopefully before node disappear
+                    status = self._task_status(self.read_pod(pod))
+                    if status == State.SUCCESS or status == State.FAILED:
+                        break
+
+            except ApiException as e:
+                # possible timeout error?
```

Comment:
I think this should log `e` in `logger`?

Reply:
I was thinking that as well. but it will create too many noises.